### PR TITLE
misc(license): Add license support

### PR DESCRIPTION
**Pull request by jdenquin**
## Description

- Add License Support

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -331,9 +331,9 @@
     t.uuid "customer_id"
     t.boolean "legacy", default: false, null: false
     t.float "vat_rate"
-    t.integer "status", default: 0, null: false
     t.bigint "credit_amount_cents", default: 0, null: false
     t.string "credit_amount_currency"
+    t.integer "status", default: 0, null: false
```
Comment (rsempe) [inline db/schema.rb:336]: You can rebuild your db to not have this change ;)

Comment (jdenquin) [reply db/schema.rb:336]: ah yes wanted to remove the `schema.rb` from this PR scope

File: spec/lib/lago_utils/license_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+RSpec.describe LagoUtils::License do
+  subject(:license) { described_class.new(url) }
+
+  let(:url) { 'https://license.lago' }
+
+  before do
+    ENV['LAGO_LICENSE'] = 'test-license'
+  end
+
+  describe '.verify' do
```
Comment (rsempe) [inline spec/lib/lago_utils/license_spec.rb:14]:
```suggestion
  describe '#verify' do
```

Comment (rsempe) [inline spec/lib/lago_utils/license_spec.rb:14]:
```suggestion
  describe '#verify' do
```

File: spec/lib/lago_utils/license_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      it 'sets premium to true' do
+        license.verify
+
+        expect(license).to be_premium
+      end
+    end
+
+    context 'when license is not present' do
+      before do
+        ENV['LAGO_LICENSE'] = nil
```
Comment (rsempe) [inline spec/lib/lago_utils/license_spec.rb:36]: Are you sure you don't have to set it again after the test?

**Review by rsempe — APPROVED** [review]

File: spec/lib/lago_utils/license_spec.rb (hunk 3 above)
Comment (jdenquin) [reply spec/lib/lago_utils/license_spec.rb:36]: since it only runs for test I pretty sure yes!

File: lib/lago_utils/lago_utils/license.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    def initialize(url)
+      @url = url
+      @premium = false
+    end
+
+    def verify
+      return if ENV['LAGO_LICENSE'].blank?
+
+      http_client = LagoHttpClient::Client.new("#{url}/verify/#{ENV['LAGO_LICENSE']}")
+      response = http_client.post({}, [])
```
Comment (lovrocolic) [inline lib/lago_utils/lago_utils/license.rb:14]: TBH I'm not sure why we don't use HTTP GET here since we just fetch some informations here

Comment (jdenquin) [reply lib/lago_utils/lago_utils/license.rb:14]: that designed in the license service this way, but we can update it with a `GET`, wdyt @rsempe ?

Comment (rsempe) [reply lib/lago_utils/lago_utils/license.rb:14]: Yep, `GET` makes more sense to me as well

Comment (jdenquin) [reply lib/lago_utils/lago_utils/license.rb:14]: let's go for a GET so!

**Pull request merged**